vcd/resource_vcd_provider_vdc.go

```diff
@@ -0,0 +1,688 @@
+			},
+			"vcenter_id": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "ID of the vCenter server that provides the resource pools and datastores",
+			},
+			// TODO: metadata handling to be added after refactoring of conflicting fields "metadata" and "metadata_entry"
+			//"metadata": {
+			//	Type:        schema.TypeMap,
+			//	Computed:    true,
+			//	Description: "Key and value pairs for Provider VDC metadata",
+			//	Deprecated:  "Use metadata_entry instead",
+			//},
+			//"metadata_entry": getMetadataEntrySchema("Provider VDC", false),
+		},
+	}
+}
+
+func resourceVcdProviderVdcCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	vcdClient := meta.(*VCDClient)
+	providerVdcName := d.Get("name").(string)
+	providerVdcDescription := d.Get("description").(string)
+	vcenterId := d.Get("vcenter_id").(string)
+	managerid := d.Get("nsxt_manager_id").(string)
+	hwVersion := d.Get("highest_supported_hardware_version").(string)
+	rawNetworkPoolIds := d.Get("network_pool_ids").(*schema.Set)
+	rawResourcePoolIds := d.Get("resource_pool_ids").(*schema.Set)
+	managerHref, err := url.JoinPath(vcdClient.Client.VCDHREF.String(), "admin", "extension", "nsxtManagers", managerid)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	vcenter, err := vcdClient.GetVcenterById(vcenterId)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	vcenterUrl, err := vcenter.GetVimServerUrl()
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	if rawNetworkPoolIds.Len() == 0 {
+		return diag.Errorf("no network pool was provided")
+	}
+	if rawNetworkPoolIds.Len() > 1 {
+		return diag.Errorf("only one network pool can be used to create a Provider VDC")
+	}
+	if rawResourcePoolIds.Len() == 0 {
+		return diag.Errorf("no resource pool was provided")
+	}
+	if rawResourcePoolIds.Len() > 1 {
+		return diag.Errorf("only one resource pool can be used to create a Provider VDC")
+	}
+	resourcePoolId := rawResourcePoolIds.List()[0].(string)
+	networkPoolId := rawNetworkPoolIds.List()[0].(string)
+
+	resourcePool, err := vcenter.GetResourcePoolById(resourcePoolId)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	defaultHwVersion, err := resourcePool.GetDefaultHardwareVersion()
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	if hwVersion == "" {
+		hwVersion = defaultHwVersion
+	}
+	rawStorageProfileNames := d.Get("storage_profile_names").(*schema.Set)
+	var wantedStorageProfiles []string
+	for _, sp := range rawStorageProfileNames.List() {
+		wantedStorageProfiles = append(wantedStorageProfiles, sp.(string))
+	}
+	if len(wantedStorageProfiles) == 0 {
+		return diag.Errorf("no storage profiles were indicated")
+	}
+
+	nsxtManagers, err := vcdClient.QueryNsxtManagerByHref(managerHref)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	if len(nsxtManagers) > 1 {
+		return diag.Errorf("more than one NSX-T manager found with ID %s", managerid)
+	}
```

Comment:
Will this not crash later in code if `0` nsx-t managers are found?

Reply:
Yes. Added a check